qutip/qip/circuit.py

```diff
@@ -183,6 +196,147 @@ def __repr__(self):
     def _repr_latex_(self):
         return str(self)
 
+    def _qasm_defn_from_resolved(self, gates_lst, qasm_out):
```

Comment:
Will it be possible to move these private methods to qasm.py instead of leaving them here in the circuit class? This class is quite heavily loaded. I see that all of these methods take the `qasm_out` object. So a better design could be making them methods of `QasmOutput` class that take a `QubitCircuit` as input, right?

Reply:
Hmm, I mostly just wanted to have the `_qasm_out` function in the circuit.py file since that makes the code quite modular (you can just call the same function for each circuit "object"). I could definitely move these functions to the qasm.py file but then I'd probably also have to move the `_qasm_out` function there which I'm not in complete favor of. What do you think?

Reply:
>I mostly just wanted to have the _qasm_out function in the circuit.py file since that makes the code quite modular (you can just call the same function for each circuit "object").

I agree with this, having a `qasm_out`-like method for Gate and Measurement is indeed a neat solution. 

What I'm thinking is more about "what is the class QasmOutput meant to do?". In particular, where should we put all definitions of `qutip_gate to qasm_gate`. At the moment, `_GATE_NAME_TO_QASM_NAME ` is located in `qasm.py`, while other definitions are in `_qasm_defns` and `_qasm_defn_resolve`. I'm more in line with having them at the same place.